pkg/kube/container_oci.go

```diff
@@ -248,35 +249,56 @@ func (t *containerTranslator) configureResources() {
 }
 
 func (t *containerTranslator) configureProcess() error {
-	const (
-		execScript = "/.singularity.d/actions/exec"
-		runScript  = "/.singularity.d/actions/run"
-	)
-	if t.cont.imgInfo.OciConfig != nil {
+	cmd := t.cont.GetCommand()
+	args := t.cont.GetArgs()
+	cwd := t.cont.GetWorkingDir()
+
+	if t.cont.imgInfo.Ref.URI() == singularity.DockerDomain && t.cont.imgInfo.OciConfig != nil {
```

Comment:
I think this condition could add an additional `&& !hasBinSh()`, this way we could fallback to the legacy singularity execution taking care of environment variables and runscript (custom or not), thought ?

Reply:
Condition `t.cont.imgInfo.Ref.URI() == singularity.DockerDomain` makes this section reachable only when image was built from docker, e.g `singularity build docker://foo/bar`. That means there are no exec/runscripts that may change oci config behavior. I don't see any profit gain here, maybe I am missing something.

Reply:
The only thing that may change is that if shell exists we will not fall with error if image has neither entrypoint nor cmd from k8s YAML, in that case we will execute `/bin/sh`.